dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -175,10 +178,82 @@ public Version parseWorkflowContent(String filePath, String content, Set<SourceF
                 validationMessageObject.put(filePath, "CWL file is malformed or missing, cannot extract metadata: " + message);
                 version.addOrUpdateValidation(new Validation(DescriptorLanguage.FileType.DOCKSTORE_CWL, false, validationMessageObject));
             }
+
         }
         return version;
     }
 
+    /**
+     * Examine the specified set of SourceFiles for CWL language versions,
+     * set the language version of each SourceFile, and set the list of
+     * language versions of the specified Version.
+     *
+     * In CWL, each SourceFile can contain multiple tools or subworkflows,
```

Comment:
Not sure how I feel about "latest", it makes more sense than earliest (hopefully workflow engines can run anything older than the latest CWL version they support) but at some point they may also drop/deprecate old versions. 
Could imagine a CWL workflow engine that only does 1.1 and above. 

But maybe this is an edge-enough case that we could wait for a round two?

Reply:
You could argue it both ways; if for some reason somebody is running an older CWL workflow engine, it may run newer versions of CWL. :) But Denis' case is probably more likely, I'm guessing. Agreed on waiting...

The other option is to bite the bullet and support multiple versions in source files.

Have we  actually seen a real CWL file with multiple versions?

Reply:
> You could argue it both ways; if for some reason somebody is running an older CWL workflow engine, it may run newer versions of CWL. :) But Denis' case is probably more likely, I'm guessing. Agreed on waiting...
> 
> The other option is to bite the bullet and support multiple versions in source files.
> 
> Have we actually seen a real CWL file with multiple versions?

It's probably pretty rare, I can't recall ever seeing one.

Reply:
Because a reasonable ordering would potentially be more useful than a random arrangement, and "latest" seems to be pretty good, or at least not worse than anything else, I left it that way for now.  It's easy to change if needed...

Reply:
Ok, but do document this choice/potential follow-up in a public ticket